jlink-gradle/src/main/kotlin/com/ryandens/jlink/JlinkAwareStartScriptGenerator.kt

```diff
@@ -1,11 +0,0 @@
-package com.ryandens.jlink
-
-import org.gradle.api.internal.plugins.DefaultTemplateBasedStartScriptGenerator
-import org.gradle.api.internal.plugins.StartScriptTemplateBindingFactory
-import org.gradle.util.internal.TextUtil
-
-class JlinkAwareStartScriptGenerator : DefaultTemplateBasedStartScriptGenerator(
```

Comment:
Rather than having a class per operating system, I think we can keep this class and simply pass the parameters necessary via constructor injection, like this:

```kotlin
class JlinkAwareStartScriptGenerator(
    lineSeparator: String,
    bindingFactory: StartScriptTemplateBindingFactory,
    startScriptTemplateFileName: String,
) : DefaultTemplateBasedStartScriptGenerator(
        lineSeparator,
        bindingFactory,
        utf8ClassPathResource(JlinkAwareUnixStartScriptGenerator::class.java, startScriptTemplateFileName),
    )
```

Note, the reason why we subclass `DefaultTemplateBasedStartScriptGenerator` is because the method `utf8ClassPathResource` is a protected method on that class, enabling us to use it easily via this method. Otherwie, I would avoid subclassing altgoether and simply create `DefaultTemplateBasedStartScriptGenerator` instances in the `JlinkJreApplicationDistributionPlugin`